### ISSUE TITLE
Nonetype error after hyperopting

### ISSUE THREAD

**Issue by snussik** [freqtrade/freqtrade#5307]
The development branch of freqtrade creates json file with best found optimized params alongside with strategy. Ive found this feature description here: #5219
Buy after that running backtest or hyperopt causes error:

```
  buy_params = deep_merge_dicts(params.get('buy', {}), getattr(self, 'buy_params', None))
  File "/home/snusik/develop/ft/freqtrade/misc.py", line 140, in deep_merge_dicts
  destination[key] = value
  TypeError: 'NoneType' object does not support item assignment
```

My json looks like this:
```
{
  "strategy_name": "FWR",
  "params": {
    "roi": {
      "0": 999.04,
      "30": 999.02,
      "60": 999.01
    },
    "stoploss": {
      "stoploss": -9.1
    },
    "trailing": {
      "trailing_stop": false,
      "trailing_stop_positive": null,
      "trailing_stop_positive_offset": 0.0,
      "trailing_only_offset_is_reached": false
    },
    "buy" : {
      "BB_length_buy": 22,
      "RSILowerLevel": 39,
      "RSI_buy_length": 80,
      "std_multi_buy": 3.0
  },
  "sell" : {
      "BB_length_sell": 58,
      "RSIUpperLevel": 73,
      "RSI_sell_length": 78,
      "res_multi": 8,
      "std_multi_sell": 1.8
  }
  },
  "ft_stratparam_v": 1,
  "export_time": "2021-07-22 06:25:50.381658+00:00"
}
```

p.s. If I delete this json everything work fine again.

**Comment by xmatthias** [comment]
I guess you can work around this by adding an empty `buy_params = {}` to the strategy.

should be pretty simple to fix though...

**Comment by xmatthias** [comment]
Closed in b84a1d0c92596f9a5fae34611da8eafe5ba4e981
